### DISCUSSION TITLE
What are the "ping_time" units for lncli listpeers (perhaps microseconds)??

### DISCUSSION THREAD

**Discussion by MWDJ** [lightningnetwork/lnd#6307] [Q&A] [answered]
When operating an LND node with Tor, I have always been curious about the ping time units reported by the various interfaces (for example lncli and RTL).  Typical times reported vary from 200000 to over 1000000+.  

Does anyone know the units for the ping_time results observed in the "lncli listpeers" command?  It appears that most other status items have a unit of measure associated with them except for ping_time.

It might make sense that they are in microseconds... 200000 = 200ms or 0.2s.  Can anyone confirm this?
Also, what is a good ping time to expect between peers (both clearnet and through Tor)?

Thank you.
-DJ

**Comment by guggero** [comment, marked as answer]
It's indeed in microseconds: https://github.com/lightningnetwork/lnd/blob/master/peer/brontide.go#L358
This should probably be documented in the API docs here: https://github.com/lightningnetwork/lnd/blob/master/lnrpc/lightning.proto#L1614
